### PR TITLE
Test Pulp process labels

### PR DESCRIPTION
**Pull request by Ichimonji10**
Add module `pulp_smash.tests.platform.cli.test_selinux`.

Fix: https://github.com/PulpQE/pulp-smash/issues/444

### CODE REVIEW

**Comment by Ichimonji10** [comment]
@dkliban, every test in this PR fails, and all the failures are alike. Here's a sample failure:

```
======================================================================
FAIL: test_celerybeat (pulp_smash.tests.platform.cli.test_selinux.ProcessLabelsTestCase)
Verify the label of the Pulp celerybeat process.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ichimonji10/code/pulp-smash/pulp_smash/tests/platform/cli/test_selinux.py", line 79, in test_celerybeat
    self._do_test('unconfined_u:system_r:celery_t:s0', 'celerybeat.pid')
  File "/home/ichimonji10/code/pulp-smash/pulp_smash/tests/platform/cli/test_selinux.py", line 45, in _do_test
    self.assertEqual(proc.label, label, procs)
AssertionError: 'system_u:system_r:celery_t:s0' != 'unconfined_u:system_r:celery_t:s0'
- system_u:system_r:celery_t:s0
? ^^^^ ^
+ unconfined_u:system_r:celery_t:s0
? ^^^^^^^^ ^
 : [Process(label='system_u:system_r:celery_t:s0', args='/usr/bin/python /usr/bin/celery beat --app=pulp.server.async.celery_instance.celery --scheduler=pulp.server.async.scheduler.Scheduler --workdir=/var/run/pulp/ -f /var/log/pulp/celery
beat.log -l INFO --detach --pidfile=/var/run/pulp/celerybeat.pid')]
```

In every case, the issue is that Pulp processes have a context starting with `unconfined_u`, not `system_u`. Can you give any insight into this?

File: pulp_smash/tests/platform/cli/test_selinux.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# coding=utf-8
+"""Tests to verify that Pulp and SELinux are interacting properly."""
```
Comment (elyezer) [inline pulp_smash/tests/platform/cli/test_selinux.py]: Maybe `Tests to verify that Pulp has the proper SELinux permissions`?

File: pulp_smash/tests/platform/cli/test_selinux.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import unittest
+from collections import namedtuple
+
+from pulp_smash import cli, config, utils
+
+
+PS_FIELDS = ('label', 'args')
+
+Process = namedtuple('Process', PS_FIELDS)
```
Comment (elyezer) [inline pulp_smash/tests/platform/cli/test_selinux.py]: namedtuple :heart:

File: pulp_smash/tests/platform/cli/test_selinux.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    def _do_test(self, label, arg):
+        """Assert that certain processes have a label of ``label``.
+
+        Select all processes that were invoked with an argument string
+        containing ``arg``. Assert that at least one such process exists, and
+        that all such processes have a label of ``label``.
+        """
+        procs = [proc for proc in self.procs if arg in proc.args]
+        self.assertGreater(len(procs), 0, arg)
```
Comment (elyezer) [inline pulp_smash/tests/platform/cli/test_selinux.py]: Thanks for checking that at least one process is running.

File: pulp_smash/tests/platform/cli/test_selinux.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        They should have a label of ``unconfined_u:system_r:httpd_t:s0``.
+        """
+        self._do_test('unconfined_u:system_r:httpd_t:s0', 'wsgi:pulp')
+
+    def test_resource_manager(self):
+        """Verify the labels of the ``resource_manager`` processes.
+
+        They should have a label of ``unconfined_u:system_r:celery_t:s0``.
+        """
+        self._do_test('unconfined_u:system_r:celery_t:s0', 'resource_manager')
```
Comment (elyezer) [inline pulp_smash/tests/platform/cli/test_selinux.py]: `unconfined_u:system_r:celery_t:s0` is being used on all tests, what about having a constant for this?

Comment (Ichimonji10) [inline pulp_smash/tests/platform/cli/test_selinux.py]: It's actually used in three of the four tests. :wink: But I'll add a constant or two.

Comment (elyezer) [inline pulp_smash/tests/platform/cli/test_selinux.py]: Ahh missed the one that does not use it. Anyway adding a constant will help maintenance, but it is not a big deal, we can decide to refactor later.

**Comment by elyezer** [comment]
LGTM, I have not added the label because there's the Needs Work assigned and I am not sure if something else is missing.

**Comment by Ichimonji10** [comment]
The issue is that the tests currently fail, due to *slightly* wrong SELinux labels. It's easy to update the test to make them pass, but I'd like to get some feedback from @dkliban on the right way to do that.

**Comment by Ichimonji10** [comment]
@dkliban Thanks for the clarification!

@elyezer The tests now pass. Only minor modifications were needed. :smile:

**Pull request merged**